src/main/java/nextstep/courses/domain/Provide.java

```diff
@@ -0,0 +1,46 @@
+package nextstep.courses.domain;
+
+import nextstep.courses.CanNotCreateException;
+import nextstep.courses.CanNotJoinException;
+import nextstep.courses.enumerate.ProvideType;
+import nextstep.payments.domain.Payment;
+
+public class Provide {
+    
+    private final ProvideType type;
+    private final ProvidePolicy policy;
+    
+    public Provide(ProvideType type) throws CanNotCreateException {
+        this(type, new ProvidePolicy());
+    }
+    
+    public Provide(ProvideType type, ProvidePolicy policy) throws CanNotCreateException {
+        validate(type, policy);
+        this.type = type;
+        this.policy = policy;
+    }
+    
+    private void validate(ProvideType type, ProvidePolicy policy) throws CanNotCreateException {
+        if(policy.isNotCorrectBetween(type)) {
+            throw new CanNotCreateException("강의타입과 정책이 일치하지 않습니다");
+        }
+    }
+    
+    public void applyPaid(EnrolledUsers enrolledUsers, Payment payment) throws CanNotJoinException {
+        if(this.type == ProvideType.FREE) {
```

Comment:
```suggestion
        if(this.type.isFree()) {
```
enum을 class와 같이 메시지를 보내는 방식으로 구현 가능